src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+        #       are also counted as properly cleaned and can trigger archival later
+
+        # Build a list of bool flags based on the mask of PlineMarkers
+        cleanFlagsList = [wflow['CleanupStatus'][key]
+                          for key in wflow['PlineMarkers']
+                          if key in wflow['CleanupStatus'].keys()]
+
+        # If no one have worked on the workflow set the clean status to false
+        if not wflow['PlineMarkers']:
+            cleanStatus = False
+        # If we have a mask longer than the list of flags avoid false positives
+        # because of the behavior explained above - `all([]) == True`
+        elif not cleanFlagsList:
+            cleanStatus = False
+        # Figure out the final value
+        else:
+            cleanStatus = all(cleanFlagsList)
+        return cleanStatus
+
+    def setClean(self, wflow):
+        """
+        A function to set the 'IsClean' flag based on the status from all the
+        pipelines which have worked on the workflow (and have put their markers
+        in the 'PlineMarkers' list)
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        wflow['IsClean'] = self._checkClean(wflow)
+        return wflow
+
+    def archive(self, wflow):
+        """
+        Move the workflow to the proper archived status after checking
+        the full cleanup status
+        :param  wflow:      A MSRuleCleaner workflow representation
+        :param  archStatus: Target status to transition after archival
+        :return wflow:
+        """
+        # NOTE: check allowed status transitions with:
+        #       https://github.com/dmwm/WMCore/blob/5961d2229b1e548e58259c06af154f33bce36c68/src/python/WMCore/ReqMgr/DataStructs/RequestStatus.py#L171
+        if not (wflow['IsClean'] or wflow['ForceArchive']):
+            msg = "Not properly cleaned workflow: %s" % wflow['RequestName']
+            raise MSRuleCleanerArchival(msg)
+
+        # Check the available status transitions before we decide the final status
+        targetStatusList = RequestStatus.REQUEST_STATE_TRANSITION.get(wflow['RequestStatus'], [])
+        self.logger.info("targetStatusList: %s", targetStatusList)
+        return wflow
+
+    def getMSOutputTransferInfo(self, wflow):
+        """
+        Fetches the transfer information from the MSOutput REST interface for
+        the given workflow.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        headers = {'Accept': 'application/json'}
+        params = {}
+        url = '%s/data/info?request=%s' % (self.msConfig['msOutputUrl'],
+                                           wflow['RequestName'])
+        mgr = RequestHandler()
+        res = mgr.getdata(url, params=params, headers=headers, ckey=ckey(), cert=cert())
+        data = json.loads(res)['result'][0]
+        TransferInfo = data['transferDoc']
```

Comment:
wrong variable name.

Reply:
Which one?

Reply:
`TransferInfo`. They should start with lower case ;)

Reply:
got it